clamav/model.py

```diff
@@ -123,3 +125,45 @@ def age(self) -> int:
 
     def state(self) -> ClamAVHealthState:
         return ClamAVHealthState(self.raw['state'])
+
+
+class ClamAVScanEventTypes(enum.Enum):
+    ERROR = 'error'
+    RESULT = 'result'
+
+
+class ClamAVError(ModelBase):
+    def status_code(self) -> int:
+        '''Return the HTTP/1.1 status code that was given by ClamAV
+        '''
+        return self.raw['code']
+
+    def message(self) -> str:
+        return self.raw['message']
+
+
+class ClamAVScanEventClient(object):
+    def __init__(self, response):
+        self.client = sseclient.SSEClient(response)
+
+    def process_events(self) -> typing.Union[ClamAVScanResult, ClamAVError]:
+        '''Process the events sent by our ClamAV service
+
+        Our ClamAV service will send exactly one event which is of one of two types:
+            1. An event of type 'error' containing an error-code (a HTTP/1.1 Status Code) and a
+                message, in case an error was encountered when scanning.
+            2. An event of type 'result' containing the scan result.
+
+        This method blocks until an event is received and then returns an instance of the
+        corresponding model class.
+        '''
+        event = next(self.client.events())
+        if event.event == ClamAVScanEventTypes.ERROR.value:
```

Comment:
instantiate enum using it's c'tor. no need to redundantly read the enum's value. also, this implicitly enforces the value be known

Reply:
I _think_ it is recommended practise to check for identity. decreases readability for me, though, so your choice.